refactor(migrator): clean up API calls, logging and exception handling

HeleneW-dot opened the pull request:
related to https://github.com/camunda/camunda-bpm-platform/issues/5308

- Moves API calls to their respective clients
  - also did some small renaming for clarity 
- Moves logs to their respective logging classes
  - also rephrased some logs for consitency, eg always referring to `legacyId [{}]` for easier debugging
- Differentiates runtime vs history exceptions for exception wrapping 
  - added a runtime/history context setting to existing ExceptionUtil since the exception is throw from usages of clients that could be used from either runtime or history migrator

tasso94 commented on core/src/main/java/io/camunda/migrator/impl/RuntimeValidator.java:
```diff
@@ -186,7 +168,9 @@ public void validateProcessInstanceState(String legacyProcessInstanceId) {
-    });
+    };
+
+    c7Client.fetchAndProcessProcessInstances(validator, legacyProcessInstanceId);
```
Yes, inline.

`fetchAndHandleProcessInstances` sounds good 👍

HeleneW-dot commented on core/src/main/java/io/camunda/migrator/impl/logging/DateVariableInterceptorLogs.java:
Actually instead of applying changes manually now I will delete my interceptorlogs classes, wait until your is merged and then accept your changes to the variableLogs class during conflict resolution

HeleneW-dot commented on core/src/main/java/io/camunda/migrator/impl/logging/DefaultVariableInterceptorLogs.java:
Will wait to fix until your PR is merged

HeleneW-dot commented on core/src/main/java/io/camunda/migrator/impl/clients/DbClient.java:
```diff
@@ -92,26 +106,22 @@ public void insert(String legacyId, Date startDate, Long entityKey, TYPE type) {
-        .pageSize(properties.getPageSize())
-        .maxCount(idKeyMapper::findSkippedCount)
+  public void fetchAndProcessSkippedProcessInstances(Consumer<IdKeyDbModel> callback) {
+    new Pagination<IdKeyDbModel>().pageSize(properties.getPageSize()).maxCount(idKeyMapper::findSkippedCount)
```
So I looked at adding the pagesize defaults but after trying some options I prefer the current approach because:
- to autowire the properties into pagination, we'd then need to inject pagination everywhere instead of creating with new() which is weird (otherwise properties are null)
- the alternative would be to add a factory but that feels overkill for just the pagination utility unless we have more fields that need defaulted

HeleneW-dot commented:
@tasso94 
Rebased and ready for another round 🎠 
I squashed the work into an initial commit and one for the PR hints + one to fix all the rebase conflicts. Might still end up a messy review though apologies for that.

I noted we have quite a large amount of nested ifs in the HistoryMigrator at the moment, I think it would be good for readability to do a small refactoring to those those (eg to return early) or extract them into eg a HistoryMigratorValidator instead. Happy to do this in a follow up if you agree

HeleneW-dot commented on core/src/main/java/io/camunda/migrator/HistoryMigrator.java:
```diff
@@ -158,363 +126,235 @@ public void start() {
+      DecisionDefinitionDbModel dbModel = decisionDefinitionConverter.apply(legacyDecisionDefinition);
+      decisionDefinitionMapper.insert(dbModel);
+      Date deploymentTime = c7Client.getDefinitionDeploymentTime(legacyDecisionDefinition.getDeploymentId());
+      dbClient.insert(legacyId, deploymentTime, dbModel.decisionDefinitionKey(), HISTORY_DECISION_DEFINITION);
```
Actually the decision definition part will be implemented "properly" with https://github.com/camunda/camunda-bpm-platform/issues/5307

tasso94 reviewed (approved):
Very nice! 👍
